lib/phoenix_live_view/html_engine.ex

```diff
@@ -762,4 +765,45 @@ defmodule Phoenix.LiveView.HTMLEngine do
   defp void?(_), do: false
 
   defp to_location(%{line: line, column: column}), do: [line: line, column: column]
+
+  # Check if `phx-update` or `phx-hook` is present in attrs and raises in case
+  # there is no ID attribute set.
+  defp validate_phx_attrs!(attrs, meta, state),
+    do: validate_phx_attrs!(attrs, meta, state, %{attr: nil, id?: false})
```

Comment:
Oh, if we need to keep a map, then instead I would use a function of arity 5, with id? and attr as arguments.